exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/impl/OptionBuilder.java

```diff
@@ -129,10 +133,60 @@ public OptionBuilder setSchema(TupleMetadata schema) {
     return this;
   }
 
+  /**
+   * Provide an optional column conversion factory. Can create a "shim"
```

Comment:
From the description is not quite clear to me why we need to both options: column conversions and schema transform. Could not we have only one that would function as both? Can you please explain a little bit more?